### PR TITLE
support for murmur3

### PR DESCRIPTION
**Pull request by pgte**
Added support for murmur3 hashing.
Uses new dependency [murmurhash3js](http://pid.github.io/murmurHash3js/).
Since output is a 32bit number, had to create a generic function to convert from number into a 32 bit buffer.
Also, since input can only be a string, converting from buffer if needed.
murmur3 can use a seed, but to get parity with go-ipfs, leaving it to the default of `0`.

### CODE REVIEW

**Comment by pgte** [comment]
Depends on https://github.com/multiformats/js-multihash/pull/30

**Comment by diasdavid** [comment]
released `js-multihash` :)

**Comment by pgte** [comment]
@diasdavid Thanks! Dep upgraded. Merge away if you will. :)

File: src/crypto.js [hunk 1]
```diff
@@ -9,6 +10,21 @@ const toBuf = (doWork, other) => (input) => {
...
+  return doWork(input, other)
+}
+
+const to32BitBuf = (doWork, other) => (input) => {
+  let number = doWork(input, other)
+  const bytes = []
+  for (let i = 0; i < 4; i++) {
+    bytes.push(number & 0xff)
+    number = number >>> 8
+  }
```
Comment (dignifiedquire) [inline src/crypto.js]: Could you test if this gets faster if you unroll this loop, given these are only 4 iterations the code amount wouldn't be too bad.

Comment (dignifiedquire) [inline src/crypto.js]: v8 might optimize this away/unroll itself but I have seen instances where loop unrolling does help a good chunk in js

**Comment by jbenet** [comment]
curious what murmur is going to be used for?

File: src/crypto.js (hunk 1 above)
Comment (pgte) [inline src/crypto.js]: Tried but unrolling the loop resulted in no gain.

**Comment by pgte** [comment]
@jbenet it's the hashing function for HAMT.

File: src/crypto.js (hunk 1 above)
Comment (dignifiedquire) [inline src/crypto.js]: too bad, but thank you for checking

**Pull request merged**